lib/ansible/modules/cloud/misc/virt_net.py

```diff
@@ -273,8 +279,46 @@ def modify(self, entryid, xml):
                         res = 0
                     if res == 0:
                         return True
-            #  command, section, parentIndex, xml, flags=0
-            self.module.fail_json(msg='updating this is not supported yet %s' % to_native(xml))
+        if new_data.tag == 'host' and new_data.findtext('hostname'):
+            ip = new_data.get('ip')
+            hosts = old_data.xpath('/network/dns/host')
+            # find the one mac we're looking for
+            host = None
+            for h in hosts:
+                if h.get('ip') == ip:
+                    host = h
+                    break
+            if host is None:
+                # add the host
+                if not self.module.check_mode:
+                    res = network.update(libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+                                         libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+                                         -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
+                else:
+                    # pretend there was a change
+                    res = 0
+                if res == 0:
+                    return True
+            else:
+                # change the host
+                if new_data.findtext('hostname') == host.findtext('hostname'):
+                    return False
+                else:
+                    if not self.module.check_mode:
+                        res = 0
+                        res += network.update(libvirt.VIR_NETWORK_UPDATE_COMMAND_DELETE,
+                                              libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+                                              -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
+                        res += network.update(libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+                                              libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+                                              -1, xml, libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT)
```

Comment:
I'm not a big fan of the use of `libvirt.VIR_NETWORK_UPDATE_AFFECT_CURRENT` here and line 296. If the user restarts the network, the change will be gone and I don't think this is what is expected. You can `libvirt.VIR_DOMAIN_AFFECT_LIVE|libvirt.VIR_DOMAIN_AFFECT_CONFIG` instead.

Reply:
It's the it's done for dhcp. And I didn't think it actually works as documented.

VIR_NETWORK_UPDATE_AFFECT_CURRENT | = | 0 (0x0) | affect live if network is active, config if it's not active
-- | -- | -- | --

That doesn't sound very reasonable to me.

Reply:
I agree. `VIR_NETWORK_UPDATE_AFFECT_CURRENT` depends on the state of the resource. This is not what an Ansible user expects IMHO. Thanks for the fix (7698fadb176587e7f4272f985aac9dc09a8a2b63).